### PR TITLE
fileexporter/receiver: refactor fileexporter and read compressed json in filereceiver

### PR DESCRIPTION
**Pull request by moh-osman3**
**Description:** 
[WIP] not working

This PR:
- refactors fileexporter to use a compressed writer
- adds support for logs and traces in filereceiver
- adds support to read compressed json in filereceiver

### CODE REVIEW

**Comment by atoulme** [comment]
Before you push too much into this, please know this PR's scope is too big for review. Please break it in 2.

File: exporter/fileexporter/file_exporter.go [hunk 1]
```diff
@@ -70,55 +69,112 @@ func (e *fileExporter) consumeTraces(_ context.Context, td ptrace.Traces) error
...
+				}
+			}()
+
+			return &lineWriter{
+				file: bufio.NewWriter(fw),
+			}
+		}
+	}
+
+	return &lineWriter{
```
Comment (jmacd) [inline exporter/fileexporter/file_exporter.go:118]: 👍 👍

File: exporter/fileexporter/file_exporter.go [hunk 2]
```diff
@@ -70,55 +69,112 @@ func (e *fileExporter) consumeTraces(_ context.Context, td ptrace.Traces) error
...
+					if fw.Flush() != nil {
+						return
+					}
+				}
+			}()
+			return &fileWriter{
+				file: bufio.NewWriter(fw),
+			}
+		}
+	}
```
Comment (jmacd) [inline exporter/fileexporter/file_exporter.go:157]: This changes compression to be per-stream not per-record, which I think is OK as long as the reader (a) has never supported this, (b) will support this after your change.

**Comment by moh-osman3** [comment]
@atoulme @jmacd Thanks for the feedback! I have now split this PR into the following 
1. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22079
2. https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/22080

**Pull request closed without merging**